### DISCUSSION TITLE
Caching? Model Caching?

### DISCUSSION THREAD

**Discussion by Yahav** [spatie/laravel-medialibrary#2650] [Q&A] [answered]
Hey

I'm interested if any of you have any experience with caching either the models associated media (mediable model) or the media model as well.
My initial thought was to cache the associated model (the mediable model) when the media relation is loaded, however, as media changes isn't picked up by the mediable observers.
Any idea what would be the best way to achieve media caching?

**Comment by Yahav** [comment, marked as answer]
To anyone interested, ended doing it like that:
[https://gist.github.com/Yahav/65e3fd68614653bc93c66d5679069ba9](https://gist.github.com/Yahav/65e3fd68614653bc93c66d5679069ba9)

**Reply by erikn69** [reply]
Maybe you can do a package for packagist, also how to cache relations on https://github.com/spatie/laravel-permission
